plotter/sankey.go

```diff
@@ -0,0 +1,440 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"fmt"
+	"image/color"
+	"math"
+	"sort"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+	"github.com/ready-steady/spline"
+)
+
+// A Sankey diagram presents stock and flow data as rectangles representing
+// the amount of each stock and lines between the stocks representing the
+// amount of each flow.
+type Sankey struct {
+	// StockPad specifies the padding between
+	// stocks in the same category, in chart units.
+	StockPad float64
+
+	// Color specifies the default fill
+	// colors for the stocks and flows. Colors can be
+	// modified for individual stocks and flows.
+	color.Color
+
+	// StockBarWidth is the widths of the bars representing
+	// the stocks. The default value is 15% larger than the
+	// height of the stock label text.
+	StockBarWidth vg.Length
+
+	// LineStyle specifies the default border
+	// line style for the stocks and flows. Styles can be
+	// modified for individual stocks and flows.
+	draw.LineStyle
```

Comment:
LineStyle and TextStyle should not really be embedded either, particularly TextStyle since it has methods.